### DISCUSSION TITLE
Патч для V380E3_C6

### DISCUSSION THREAD

**Discussion by didenkoae** [Arkady23/V380-Web#41] [Ideas]
Доброго дня Аркадий!
Есть возможность сделать патч для камеры с данными:
"{"soft_id":"V380E3_C6","dev_id":53355493,"version":3040202,"timestamp":1652261848}"
Версия ПО: AppV380E21_V3.4.2.2_20210428
Версия прошивки: HwV380E21_WF3_PTZ_WIFI_20201212
?
Заранее спасибо!

**Comment by Arkady23** [comment]
Что-то последнее время мои патчи не устанавливаются. Попробуйте
[httpd_V380E3_C6_9090909_v20210620.zip](https://github.com/Arkady23/V380-Web/files/8669086/httpd_V380E3_C6_9090909_v20210620.zip)

**Reply by didenkoae** [reply]
Да, не установился(((
А вот v20210620 роли не играет? у меня 20210428 или 20201212, я не понял куда v относится

**Reply by Arkady23** [reply]
Да, это роли не играет. Когда придет обновление, не устанавливайте. Попробуйте его скачать по ссылке из трафика. Мне интересно разобраться, что они поменяли.

**Reply by didenkoae** [reply]
Мне еще ниразу с октября не прилетали обновы на камеру, только на приложение.

**Reply by didenkoae** [reply]
Скачать обнову и вам скинуть?

**Reply by Arkady23** [reply]
Да, обязательно. Если получится скачать, скачайте. Еще с новыми версиями не было обновлений. Что-то поменяли.

**Reply by didenkoae** [reply]
А ваш патч качает архивы из инета, эти архивы доступны? может попробовать их на флешку накидать? папка cgi-bin должна создаваться до скачивания? могу поработать вашим тестером.

**Reply by Arkady23** [reply]
Раньше патч анализировался только на hwname, он же soft_id. Содержание патчей у всех камер я делаю одинаково. Просто меняю код hwname. Если вы знаете свой, то все подряд устанавливать нет смысла.

**Reply by didenkoae** [reply]
Не, я про саму работу баш-скриптов, они качают какой-то мини-апач с инета, собирают его, запускают, верно? на самой камере урезанный линух стоит?
я не смог телнел на ней запустить, ни через ceshi.ini, ни через патчи с 4pda((
а ssh можно запускать на ней?

**Reply by Arkady23** [reply]
Только телнет. Я про такое не знаю, что что-то с инета грузится. Если патч грузится, то приходит сообщение о том, что вышел новый патч (равно —  обновление). И по вашему разрешению грузится и устанавливается. Линукс стоит. Да он не полный. Места и памяти мало.

**Reply by didenkoae** [reply]
Что делают файлы 
install_20210620,
startup.sh,
update.sh,
6a1729e4b6ec6b332d2320af852d928d.patch?
я про ваш патч, я не про обнову на камеру от официалов

**Reply by Arkady23** [reply]
startup.sh - это автозагрузка при включении камеры. Что туда поместишь, то и работает.
update.sh - это должна быть распаковка архива install_20210620 и удаление ненужных файлов, корректировка startup.sh, чтобы второй раз не пыталась больше распаковывать.
6a1729e4b6ec6b332d2320af852d928d.patch - это закодированный по алгоритму обновлений официального разработчика мой командный файл, который прописывает в системе запуск startup.sh при старте системы, если он существует.

Я так подозреваю, что алгоритм обновлений официального разработчика поменялся.

**Reply by didenkoae** [reply]
А как же веб-сервер? он уже есть на камере, и командный файл лишь включает его? но это нереально, т.к. верстка, язык, подгрузка vlc отсутсвует на камере.

**Reply by Arkady23** [reply]
Вэб-сервер в архиве install_20210620. Он распаковывается на sd, т.к. в камере места нет.
Vlc и ничего другого в install_20210620 нет. Vlc должен быть в вашем компьютере.

**Reply by didenkoae** [reply]
А почему низя просто все необходимые файлы скинуть на флешку, что б никакой распаковки из архивов не было?

**Reply by Arkady23** [reply]
Раньше так было, но у некоторых на ПК вирусы и они заражают вэб-сервер. Файлы при копировании портились, изменялась длина файлов. Поэтому пришлось сделать запаковку по максимуму в tar-архив.

**Reply by didenkoae** [reply]
Пичалька... теперь ждать обнову на камеру? а если она не прилетит, то фсё?

**Reply by didenkoae** [reply]
а какие логин-пасс на камеру по телнету?

**Reply by Arkady23** [reply]
Никто не знает. Подберете — сообщите.

**Reply by didenkoae** [reply]
Мне китайцы патч прислали "AKV380_3918E-V300__V3.5.1.1_2022-04-21". Скинуть?

**Reply by Arkady23** [reply]
Хорошо. Давай. Только проверь. Он выполняется или нет. Рабочий или нет.

**Reply by didenkoae** [reply]
Стало:
Software: AppV380E21_V3.5.1.1_20210428
Firmware: HwV380E21_WF3_PTZ_WIFI_20201212
Было:
Версия ПО: AppV380E21_V3.4.2.2_20210428
Версия прошивки: HwV380E21_WF3_PTZ_WIFI_20201212

**Reply by didenkoae** [reply]
Куда архив скинуть? на почту? тут я не нашел возможность добавления файла

**Reply by Arkady23** [reply]
Просто перетаскивается файл в сообщение и отправить.

**Reply by didenkoae** [reply]
[AKV380_3918E-V300__V3.5.1.1_2022-04-21.zip](https://github.com/Arkady23/V380-Web/files/8675731/AKV380_3918E-V300__V3.5.1.1_2022-04-21.zip)

**Reply by Arkady23** [reply]
Интересный патч. Мне кое-что не понятно. Наверное, нужно будет делать разные варианты и пробовать, пока не получится. Если вы согласны пробовать, то тогда можно понять в чем дело. Я не сейчас, а в течении нескольких часов выложу вариант.

**Reply by didenkoae** [reply]
Согласен! жду

**Reply by Arkady23** [reply]
[V380E3_C6_test1.zip](https://github.com/Arkady23/V380-Web/files/8679081/V380E3_C6_test1.zip)

**Reply by Arkady23** [reply]
[V380E3_C6_test2.zip](https://github.com/Arkady23/V380-Web/files/8679107/V380E3_C6_test2.zip)

**Reply by Arkady23** [reply]
Мне больше test2 нравится.

**Reply by didenkoae** [reply]
test1 не сработал

**Reply by didenkoae** [reply]
test2 тоже. папки bin и cgi не создаются

**Reply by Arkady23** [reply]
А исходные файлы удаляются? Вот эти:
updatepatch/
local_update.conf
update_record
patch_reuse

**Reply by didenkoae** [reply]
не, всё остается на своих местах.

**Reply by Arkady23** [reply]
[V380E3_C6_test3.zip](https://github.com/Arkady23/V380-Web/files/8679448/V380E3_C6_test3.zip)

**Reply by Arkady23** [reply]
[V380E3_C6_test4.zip](https://github.com/Arkady23/V380-Web/files/8679451/V380E3_C6_test4.zip)

**Reply by didenkoae** [reply]
Не взлетает(( а камера должна перезагружаться после применения патча?
как у меня этот процесс организован:
выключил камеру, достал флеху, подключил флеху к пк, удалил все файлы на ней, залил файлы с архива, воткнул в камеру и жду перезагрузку. 
если больше 5 минут ничего не происходит - отключаю камеру, достаю флеху и смотрю через пк создались ли на флешке папки bin и cgi.
флешка в exFAT.

**Reply by Arkady23** [reply]
Флэшка должна быть только в Fat32
остальное вроде правильно, если я понял некоторые слова типа "залил", как извлек из архива и скопировал.

**Reply by didenkoae** [reply]
Последние 2 архива только на exfat копировал. счас пробую на фат32.
"залил" - извлек из архива и скопировал на флешку.

**Reply by didenkoae** [reply]
fat32 не помог - нужные папки не создаются, ненужные не удаляются.

**Reply by Arkady23** [reply]
Буду думать.... не знаю пока в чем дело.

**Reply by didenkoae** [reply]
бодрого утра! а если прошить камеру более старой прошивкой - той, где патч вебсервера работал?

**Reply by Arkady23** [reply]
Похоже дело в библиотеках Линукса. ОС в камерах поменялась.

**Reply by Arkady23** [reply]
Получил официальный новый патч. На его основе сделал еще один вариант для вашей камеры. Установите и обязательно сообщите о результате. Проверьте не только http-сервер, но и telnet. Telnet то должен заработать (если раскомментировать соответствующую строку в стартовом файле на sd).
[httpd_V380E3_C6_9090909_v20210620.zip](https://github.com/Arkady23/V380-Web/files/9176901/httpd_V380E3_C6_9090909_v20210620.zip)

**Reply by Arkady23** [reply]
Переделал..... пробуйте. По крайней мере telnet то должен работать, если раскомментировать соответствующую строку в запускном файле startup.sh на sd.
[httpd_V380E3_C6_9090909_v20210620.zip](https://github.com/Arkady23/V380-Web/files/9176924/httpd_V380E3_C6_9090909_v20210620.zip)

**Reply by didenkoae** [reply]
Доброго дня! сегодня вечером или завтра попробую. О результатах отпишусь.

**Comment by mjdelima** [comment]
I have the same v380 camera as well
The httpd_V380E3_C6_9090909_v20210620.zip does not install.

The CCTV reboots after the fresh sdcard installed,
But nothing has changed after the reboot

Edit:
After emailing the V380 support, they give me a patch file:
**AKV3918E-V300__P3_V3.4.2.2__2021-04-22.zip**
[AKV3918E-V300__P3_V3.4.2.2__2021-04-22.zip](https://github.com/Arkady23/V380-Web/files/9130864/AKV3918E-V300__P3_V3.4.2.2__2021-04-22.zip)

**Reply by Arkady23** [reply]
Your patch:
[httpd_V380E3_C5_9090909_v20210620.zip](https://github.com/Arkady23/V380-Web/files/9176916/httpd_V380E3_C5_9090909_v20210620.zip)

**Comment by didenkoae** [comment]
Робит. только ползунок "RTSP и ONVIF" и "Формат плейлистов" не сохраняет
позиции после сохранения,  хотя ртсп работает

пн, 25 июл. 2022 г. в 02:22, Arkady23 ***@***.***>:

**Reply by Arkady23** [reply]
SD как отформатирована? Fat32 или что-то другое?

**Comment by didenkoae** [comment]
отформатирована в exfat

ср, 27 июл. 2022 г. в 11:53, Arkady23 ***@***.***>:

**Reply by Arkady23** [reply]
Сохраните содержимое sd на ПК и переформатируйте карту в fat32 утилитой отсюда http://ridgecrop.co.uk/index.htm?guiformat.htm
А потом верните содержимое на карту.

**Comment by didenkoae** [comment]
Ничего не изменилось. Все так же ползунок не правильно отображается и
плейлист в дефолтном положении

ср, 27 июл. 2022 г. в 13:00, Arkady23 ***@***.***>:

**Reply by Arkady23** [reply]
Когда настройки изменяются появляется файл opts.ini. Появился такой файл в папке ark-add-on?
Мне уже кто-то писал, что у него скрипты не выполняются.

**Comment by didenkoae** [comment]
появился, внутри "app=MPC"

ср, 27 июл. 2022 г. в 14:09, Arkady23 ***@***.***>:

**Reply by Arkady23** [reply]
Попробуйте поменять исполняемый файл на прилагаемый. Он вообще-то привязан под конкретный линукс, поэтому я его когда-то заменил на отвязанный.
[httpd.zip](https://github.com/Arkady23/V380-Web/files/9203650/httpd.zip)

**Comment by didenkoae** [comment]
[image: image.png]
всё то же самое...

чт, 28 июл. 2022 г. в 00:25, Arkady23 ***@***.***>:

**Comment by didenkoae** [comment]
но если в opts.ini заменить "app=VLC" - то в веб-морде отображается норм.
как там же кнопку rtsp запустить?

чт, 28 июл. 2022 г. в 08:50, Ar Dee ***@***.***>:

**Reply by Arkady23** [reply]
Если точнее, что такое "отображается норм."? Отображается, что выбран вариант VLC?

Попробуйте этот сервер от другой камеры
[httpd2.zip](https://github.com/Arkady23/V380-Web/files/9206214/httpd2.zip)

**Comment by Arkady23** [comment]
Добавил описание для установки web-сервера вручную (для новых камер).

**Comment by didenkoae** [comment]
Вроде работает! А можно добавить пакет curl для отправки уведомлений в
телегу?

ср, 3 авг. 2022 г. в 18:40, Arkady23 ***@***.***>:

**Reply by Arkady23** [reply]
Я не совсем понял после чего стало работать?

**Reply by Arkady23** [reply]
Web работает?

**Comment by didenkoae** [comment]
веб не  работал с последней версией httpd, что вы высылали. после я всё
затер на флешке и залил файлики с гитхаба - всё запустилось, даже фтп. уже
написал "скрипт" для выгрузки записанных файлов на ФТП-сервер, теперь хочу
попробовать в телегу слать, но для этого нужен curl, т.к. на нем у меня уже
подняты уведомлялки с серверов.

вт, 9 авг. 2022 г. в 14:48, Arkady23 ***@***.***>:

**Reply by Arkady23** [reply]
С моего дистрибутива на гитхабе?

**Reply by Arkady23** [reply]
А ссылку можете уточнить на файлики с гитхаба? Что-то я запутался. Люди пишут, что не работают скрипты httpd.

**Comment by didenkoae** [comment]
Кстати да, интерфейс новый - ctrl+F5 творят чудеса.
ссылка на архив -
https://github.com/Arkady23/V380-Web/releases/download/First-time-installer/httpd_V380_any_manual_v20220805.zip
ссылка на обнову -
https://github.com/Arkady23/V380-Web/releases/download/20220805/update_20220805.zip
на фтп заливает, осталось курлом в телегу слать. пытался подсадить с
центоса - естественно не взлетело...

вт, 9 авг. 2022 г. в 15:49, Ar Dee ***@***.***>:

**Reply by Arkady23** [reply]
Это вы WarIock?

**Reply by Arkady23** [reply]
В меню web есть пункт "Info". Если я вас правильно понял, то там пусто, ничего не отображается..... Попробуйте мое исправление пока только в info. Замените по ftp исполняемый скрипт info.sh в папке cgi-bin на sd тем что прилагаю здесь и попробуйте еще раз отобразить Info.
[info.zip](https://github.com/Arkady23/V380-Web/files/9299044/info.zip)

**Reply by Arkady23** [reply]
Если Информация не отобразится (надо ждать какое-то время, т.к. бывает отображается с большой задержкой), то попробуйте заменить этот скрипт на прилагаемый в этом сообщении.
[info2.zip](https://github.com/Arkady23/V380-Web/files/9299100/info2.zip)

**Reply by vladislav-orlovskiy** [reply]
> Кстати да, интерфейс новый - ctrl+F5 творят чудеса. ссылка на архив - https://github.com/Arkady23/V380-Web/releases/download/First-time-installer/httpd_V380_any_manual_v20220805.zip ссылка на обнову - https://github.com/Arkady23/V380-Web/releases/download/20220805/update_20220805.zip на фтп заливает, осталось курлом в телегу слать. пытался подсадить с центоса - естественно не взлетело... вт, 9 авг. 2022 г. в 15:49, Ar Dee ***@***.***>:
> […](#)
> Установка 1. Отформатировать sd-карту в FAT32 (Вот эта программа <http://ridgecrop.co.uk/index.htm?guiformat.htm> подходит). <[https://github.com/Arkady23/V380-Web#%D0%B4%D0%BB%D1%8F-%D0%BA%D0%B0%D0%BC%D0%B5%D1%80-%D1%81-%D0%B2%D0%B5%D1%80%D1%81%D0%B8%D1%8F%D0%BC%D0%B8-%D0%BF%D1%80%D0%BE%D1%88%D0%B8%D0%B2%D0%BE%D0%BA-%D0%BF%D0%BE%D1%81%D0%BB%D0%B5-2019-%D0%B3%D0%BE%D0%B4%D0%B0-%D1%83%D1%81%D1%82%D0%B0%D0%BD%D0%BE%D0%B2%D0%BA%D0%B0-%D0%B2%D1%80%D1%83%D1%87%D0%BD%D1%83%D1%8E>Для](https://github.com/Arkady23/V380-Web#%D0%B4%D0%BB%D1%8F-%D0%BA%D0%B0%D0%BC%D0%B5%D1%80-%D1%81-%D0%B2%D0%B5%D1%80%D1%81%D0%B8%D1%8F%D0%BC%D0%B8-%D0%BF%D1%80%D0%BE%D1%88%D0%B8%D0%B2%D0%BE%D0%BA-%D0%BF%D0%BE%D1%81%D0%BB%D0%B5-2019-%D0%B3%D0%BE%D0%B4%D0%B0-%D1%83%D1%81%D1%82%D0%B0%D0%BD%D0%BE%D0%B2%D0%BA%D0%B0-%D0%B2%D1%80%D1%83%D1%87%D0%BD%D1%83%D1%8E>%D0%94%D0%BB%D1%8F) камер с версиями прошивок после 2019 года (установка вручную) 1. Записать в корень карты содержимое архива <https://github.com/Arkady23/V380-Web/releases/download/First-time-installer/httpd_V380_any_manual_v20220805.zip> в составе: - ark-add-on - quick_check.ini - setup.sh 1. Вставить карту в камеру и включить её... ждать пару минут. Камера включется с запущенным сервером telnet. После чего нужно зайти через telnet по адресу камеры (логин root, пароль посмотреть в файле quick_check.ini). 2. Ввести команду через telnet: /mnt/sdcard/setup.sh 3. Камера должна перезагрузится. После чего можно заходить с помощью обозревателя интернета на адрес камеры. вот так. потом накатил обнову, но отвалился веб, включил его через телнет. потом еще что-то было, вроде починил. правда веб стоит видимо старый, ибо нет пункта включения ФТП, хотя сервер ФТП работает. 4. Честно говоря я сам уже забыл что делал... 5. [image: image.png]в 6. вот так у меня сейчас выглядит веб вт, 9 авг. 2022 г. в 15:45, Arkady23 ***@***.***>: > А ссылку можете уточнить на файлики с гитхаба? Что-то я запутался. Люди > пишут, что не работают скрипты httpd. > > — > Reply to this email directly, view it on GitHub > <[#41 (reply in thread)](https://github.com/Arkady23/V380-Web/discussions/41#discussioncomment-3357718)>, > or unsubscribe > <https://github.com/notifications/unsubscribe-auth/AZDZNZNMOULAIO4CLCHSRRTVYJHGLANCNFSM5VUK4QMA> > . > You are receiving this because you authored the thread.Message ID: > ***@***.***> >

привет. а как ты отлавливаешь триггер для уведомления?

**Reply by didenkoae** [reply]
Уведомлений о новых записях? просто раз в минуту ищу файлы *.avi.
Я не храню их на карте - сразу сливаю на ФТП, т.ч. за минуту у меня только 1 файл может создаться.

**Reply by vladislav-orlovskiy** [reply]
> Уведомлений о новых записях? просто раз в минуту ищу файлы *.avi. Я не храню их на карте - сразу сливаю на ФТП, т.ч. за минуту у меня только 1 файл может создаться.

 Можешь пожалуйста сбросить скрипт?)

**Comment by didenkoae** [comment]
вот так выглядит инфо. у меня камера по eth0, пришлось менять в конфиге))
[image: image.png]
я на 4пда скорее чтец))


ср, 10 авг. 2022 г. в 13:19, Arkady23 ***@***.***>:

**Comment by didenkoae** [comment]
https://cam.joxnet.keenetic.pro/

ср, 10 авг. 2022 г. в 14:09, Ar Dee ***@***.***>:

**Comment by didenkoae** [comment]
Не, эт не я

ср, 10 авг. 2022 г. в 13:01, Arkady23 ***@***.***>:

**Comment by didenkoae** [comment]
а что по curl? реально внедрить? далее с телегой проще будет

ср, 10 авг. 2022 г. в 14:10, Ar Dee ***@***.***>:

**Reply by Arkady23** [reply]
есть wget

**Reply by Arkady23** [reply]
Так у вас всё работает, как я понял.

**Reply by Arkady23** [reply]
Поместите в bin на sd-карте
[wget.zip](https://github.com/Arkady23/V380-Web/files/9299725/wget.zip)

**Comment by didenkoae** [comment]
Всё, что нужно мне - работает))
Сейчас буду пробовать wget

ср, 10 авг. 2022 г. в 14:46, Arkady23 ***@***.***>:

**Comment by didenkoae** [comment]
wget: note: TLS certificate validation not implemented
wget: TLS error from peer (alert code 40): handshake failure
wget: error getting response: Connection reset by peer

ср, 10 авг. 2022 г. в 15:07, Ar Dee ***@***.***>:

**Comment by didenkoae** [comment]
--no-check-certificate не помогает


ср, 10 авг. 2022 г. в 15:28, Ar Dee ***@***.***>:

**Reply by Arkady23** [reply]
Как что-то получится — напишите.

**Comment by didenkoae** [comment]
curl не будет? сложно? бессмысленно?

чт, 11 авг. 2022 г. в 02:01, Arkady23 ***@***.***>:

**Comment by didenkoae** [comment]
Какая версия wget? если он старый - https не сработает

чт, 11 авг. 2022 г. в 08:33, Ar Dee ***@***.***>:

**Reply by Arkady23** [reply]
видимо старый

**Reply by Arkady23** [reply]
Точно заметили. ftpput есть в камере. И кстати wget тоже есть в камере. Может в камере новая версия?

**Comment by didenkoae** [comment]
У меня нет((
***@***.*** ~]$ ls -l /bin | grep wget
***@***.*** ~]$ ls -l /sbin | grep wget
***@***.*** ~]$ ls -l /mnt/sdcard/bin | grep wget
-rwxr-xr-x    1 root     root        522840 Aug 10 12:07 wget

чт, 11 авг. 2022 г. в 09:57, Arkady23 ***@***.***>:

**Comment by didenkoae** [comment]
не запустился(

чт, 28 июл. 2022 г. в 09:50, Arkady23 ***@***.***>:

**Comment by didenkoae** [comment]
много воды...
руки не доходят привести в порядок.
или просто лень))

пт, 4 нояб. 2022 г. в 18:41, vladislav-orlovskiy ***@***.***>:
